### PR TITLE
10501 add widget added event

### LINKED ISSUE #10501: Track 'Added widget' event
We'd like to monitor how users use widgets in their dashboards.

This new segment event should include (if possible) all the common event attributes in addition to these specific ones:
- vis_id: id of the derived visualization
- widget_type: type of the widget (histogram, ...)
- ~rows: number of rows in dataset (in the moment of widget creation)~

cc'ed @xavijam

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes #10501

This PR adds a new event: Created widget. This event is fired every time the widgets controller successfully creates a new widget.

## Acceptance info

You will have to test all the widget operations: create, delete and update. Check everything works as expected.

You need to check the event arrives at Segment (ask @gfiorav). The event is supposed to carry at least vis_id and widget_type (along with other info).

### CODE REVIEW

**Comment by juanignaciosl** [comment]
Tests are not passing, moving this back to development:

```
16:47:42      NoMethodError:
16:47:42        private method `report!' called for #<Carto::Tracking::Events::ExportedMap:0x00000008ba16b8>
```

**Comment by gfiorav** [comment]
thanks @juanignaciosl 

I made a method private too quickly! (it was used by the controller).

File: lib/carto/tracking/validators/widget.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# encoding: utf-8
+
+module Carto
+  module Tracking
+    module Validators
+      module Widget
+        module Existence
+          def check_widget_exists!
```
Comment (juanignaciosl) [inline lib/carto/tracking/validators/widget.rb]: Where is this called from? I guess that it's some kind of meta, but as there aren't similar validators at `lib/carto/tracking/validators` I prefer checking it.

Comment (gfiorav) [inline lib/carto/tracking/validators/widget.rb]: Total meta. Any method starting with `check_` is considered a method that needs to be run and will raise something if validation doesn't pass.

Comment (gfiorav) [inline lib/carto/tracking/validators/widget.rb]: P.D.: There is a test that checks this :)

Comment (juanignaciosl) [inline lib/carto/tracking/validators/widget.rb]: You shouldn't need to see a class implementation in order to extend it. And, as this is Ruby meta, the only easy way to know that `check_` methods can be implemented is documenting it. Anyway, this certainly doesn't belong to this PR, so I'll move it to Acceptance.

Comment (gfiorav) [inline lib/carto/tracking/validators/widget.rb]: > this certainly doesn't belong to this PR

Wait, I'll add the comment doc.

Comment (gfiorav) [inline lib/carto/tracking/validators/widget.rb]: Done, see if you like it @juanignaciosl

Comment (juanignaciosl) [inline lib/carto/tracking/validators/widget.rb]: 👍

**Pull request merged**